independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/tools/ToolsUtils.java

```diff
@@ -158,11 +154,18 @@ public static ExtensionCatalog resolvePlatformDescriptorDirectly(String bomGroup
                                 continue;
                             }
                         }
+
+                        final OriginPreference originPreference = new OriginPreference(1, 1, 1, ++memberIndex, 1);
+                        Map<String, Object> metadata = memberCatalog.getMetadata();
+                        if (metadata.isEmpty()) {
+                            metadata = new HashMap<>();
```

Comment:
Why create a new instance if the metadata is empty already?

Reply:
https://github.com/quarkusio/quarkus/blob/main/independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/json/JsonExtensionCatalog.java#L88
I should probably initialize it to `new HashMap<>(0)` in the JsonExtensionCatalog and remove this ugliness.